### ISSUE TITLE
Prod-only Campaign issue: 'index-8ae7573d.js:118 ReferenceError: newPC is not defined'

### ISSUE THREAD

**Issue by MaxWilson** [MaxWilson/Arena#19]

**Comment by MaxWilson** [comment]
Looks like it's possible for JSX to refer to a piece of code that gets optimized away at runtime:

```
        let randomName() = ["Bob"; "Lea"; "Lyron"; "Mortimer"; "Sally"; "Samantha"; "Sven"; "Tyrone"] |> chooseRandom
        let newPC _ = setDraft (Some $"{randomName()}: ST 10")
        JSX.jsx """
        <div>
            <b>Party picker</b>
            <div>
                <input type="text" placeholder="Filter by name" />
                <button onClick={newPC}>New PC</button>
                <button>New monster</button>
                <button>Clear</button>
                {partyDisplay}
            </div>
        </div>
        """
```

newPC doesn't exist in the jsx when it's built for PROD, although it does exist when run locally via startAll.cmd.

**Comment by MaxWilson** [comment]
Filed as external issue: https://github.com/fable-compiler/fable/issues/3570

~~Verified that the workaround works in prod.~~ The workaround of using JavaScript syntax at the call site doesn't crash but it isn't correct either. It executes the onClick handler too soon, at render rather than on click. I don't currently know of a workaround other than downgrading to an earlier version of fable or just avoiding JSX syntax.

So I changed JSX element back to ReactElement where onClick is needed.

Closing this issue as completed.
